pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -504,6 +508,16 @@ func isRegionValid(region string) bool {
 	return false
 }
 
+// Derives the region from a valid az name.
+// Undefined if az not valid
+func azToRegion(az string) string {
+	if len(az) < 1 {
+		return ""
```

Comment:
I'm not a big fan of returning relatively innocuous empty strings for what are essentially invalid inputs.  You've used this pattern in several places.  Would it not be better to explicitly return an error (or nil)?  I'd rather have the client obviously error out.